main/src/library/Core/FromString.flix

```diff
@@ -0,0 +1,53 @@
+/*
+ *  Copyright 2020 Matthew Lutze
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+namespace FromString {
+    class FromString[a] {
+        pub def fromString(x: String): Result[a, String]
```

Comment:
I think we should change the return type to `Option[a]` per previous discussion.

Reply:
```
///
/// Optionally returns the value associated with the given string `s`.
///

Reply:
Should we change the respective library functions as well?